rasterio/_crs.pyx

```diff
@@ -49,9 +50,19 @@ class _CRS(UserDict):
         cdef int retval
 
         try:
-            # return False immediately if either value is undefined
-            if not (self and other):
+          if not self and not other:
+              warnings.warn("Comparing empty CRS Objects. Will return True")
+
+          # use dictionary equality rules first
+          if UserDict(self.data) == UserDict(other):
+              return True
+          else:
+            if not self and not other:
+                return True
```

Comment:
@vincentsarago lines 59-61 can be deleted and we can `return True` immediately after the warning at line 54.

Reply:
👌 I think I add the `UserDict` check because it was also in the PR from Matt so maybe he can explain why he added it 👋 @perrygeo

Reply:
@vincentsarago not sure I follow you. My original PR used slightly different logic 

https://github.com/mapbox/rasterio/blob/ef95e917aba155d74cb9f742376f2ddd4bb5f0bf/rasterio/_crs.pyx#L50-L60

Reply:
sorry @perrygeo I was talking about https://github.com/mapbox/rasterio/blob/ef95e917aba155d74cb9f742376f2ddd4bb5f0bf/rasterio/_crs.pyx#L50-L52 where you added pure dictionary equality test before doing gdal crs tests

Reply:
The empty CRS case is specific example of a more general problem - when you compare equality of dictionary-like objects, they should behave like dictionaries. Rather than try to hardcode the empty CRS as a special case, testing dict equality fixes all possible cases.

If the dicts are not equal, only then do we invoke OSR to find if they are equivalent projection defs.

Reply:
Excellent Thanks @perrygeo

Reply:
@vincentsarago so it looks like my change request is still valid, right?